examples/tutorials/analysis-3d/non-detection.py

```diff
@@ -0,0 +1,346 @@
+from gammapy.estimators import FluxPointsEstimator, ExcessMapEstimator
+from gammapy.modeling import Fit, select_nested_models
+from gammapy.modeling.models import (
+    PointSpatialModel,
+    PowerLawSpectralModel,
+    SkyModel,
+    ExpCutoffPowerLawSpectralModel,
+    create_crab_spectral_model,
+)
+from gammapy.stats.utils import ts_to_sigma, sigma_to_ts
+import numpy as np
+
+
+######################################################################
+# Check setup
+# -----------
+#
+
+from gammapy.utils.check import check_tutorials_setup
+
+check_tutorials_setup()
+
+
+######################################################################
+# When we don’t detect a source
+# -----------------------------
+#
+# As usual, let’s start with some general imports…
+#
+
+
+######################################################################
+# Now, load and inspect the dataset. For computational purposes, we have
+# already created a `MapDataset` from observation id 20275 from the
+# public H.E.S.S. data release and stored it in `GAMMAPY_DATA`
+#
+
+dataset = MapDataset.read("$GAMMAPY_DATA/empty_dl4/empty-dl4.fits.gz")
+dataset.peek()
+plt.show()
+
+
+######################################################################
+# We will first use the `ExcessMapEstimator` for a quick check to see if
+# there are any hotspots in the field. You may also use the
+# `TSMapEstimator`.
+#
+
+estimator = ExcessMapEstimator(sum_over_energy_groups=True, selection_optional="all")
+res1 = estimator.run(dataset)
+res1.sqrt_ts.plot(add_cbar=True)
+plt.show()
+
+
+######################################################################
+# The significance map looks rather flat! Suppose we were expecting a
+# source at the centre of our map. Lets try see if we can fit a point
+# source there.
+#
+
+spectral = PowerLawSpectralModel()
+spatial = PointSpatialModel(frame="icrs")
+spatial.lon_0.value = 187.0
+spatial.lat_0.value = 2.6
+spatial.lat_0.min = 1.0
+spatial.lat_0.max = 4.0
+spatial.lon_0.min = 185
+spatial.lon_0.max = 187
+
+skymodel = SkyModel(spatial_model=spatial, spectral_model=spectral, name="test")
+
+dataset.models = skymodel
+fit = Fit()
+res = fit.run(dataset)
+print(res.models)
+
+
+######################################################################
+# We can see that there is a slight negative excess in the centre, and
+# thus, the fitted model has a negative amplitude. We can use the
+# `select_nested_models` function to perform a likelihood ratio test to
+# see if this is significant.
+#
+
+LLR = select_nested_models(
+    Datasets(dataset), parameters=[skymodel.parameters["amplitude"]], null_values=[0]
+)
+print(LLR)
+
+
+######################################################################
+# You can see that the `ts ~ 4.7`, thus suggesting that the observed
+# fluctuations are not significant over the background. Now, we will
+# estimate the differential upper limits of the source.
```

Comment:
I propose to add the computation of the integral UL. When we have no detection, this is common to give this number